### PR TITLE
Add models

### PR DESCRIPTION
**Pull request by halitbatur**
# Create DB models

- Created DB models.
- Implement a discriminate function for the user model.

NOTE: This PR is made by @ZuluNovember, I had to revert the old PR and create a new one since it was not approved by @Shrreya

### CODE REVIEW

File: src/models/user.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      type: String,
+      required: true,
+    },
+    birthDate: {
+      type: Date,
+      required: true,
+    },
+    skills: {
+      type: [String],
+    },
```
Comment (Shrreya) [inline src/models/user.js:37]: If `skills` are optional then consider adding a `default` property as empty array.

File: src/models/user.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      required: true,
+    },
+    lastName: {
+      type: String,
+      required: true,
+    },
+    birthDate: {
+      type: Date,
+      required: true,
+    },
```
Comment (Shrreya) [inline src/models/user.js:34]: Following up from @ibraheemhajjar's suggestion in the previous PR, can you create a virtual field for `age` on this model?

**Review by Shrreya — CHANGES_REQUESTED** [review]
Great work overall! Specially setting up the discriminator schema and finding the right use for embedding and referencing.

I have a bunch of suggestions for improvement. Hope this helps. 🙂

**Review by Shrreya — APPROVED** [review]
@ZuluNovember Impeccable work! 🚀

**Review by ibraheemhajjar — APPROVED** [review]
good job @ZuluNovember

**Pull request merged**